### PR TITLE
[WIP] Adding leaderboard page

### PR DESCRIPTION
**Pull request by c-o-l-o-r**
wip

### CODE REVIEW

File: src/public-modules/Leaderboard/sagas.js [hunk 1]
```diff
@@ -6,12 +6,24 @@ import { LIMIT, PLATFORM } from './constants';
...
   try {
-    let endpoint = `leaderboard/${leaderboardCategory}?limit=${LIMIT}&platform__in=${PLATFORM}`;
-    const leaderboard = yield call(request, endpoint, 'GET');
-    yield put(loadLeaderboardSuccess(leaderboard));
+    const issuer_leaderboard = yield call(
+      request,
+      'leaderboard/issuer/',
+      'GET'
+    );
+    const fulfiller_leaderboard = yield call(
```
Comment (villanuevawill) [inline src/public-modules/Leaderboard/sagas.js]: @c-o-l-o-r you should make these calls in paralell.  You can do this with redux saga. :)

File: src/stories/LeaderCard.stories.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import { storiesOf } from '@storybook/react';
+import { action } from '@storybook/addon-actions';
+
+import moment from 'moment';
+
+import { Text } from 'components';
+import { LeaderCard } from 'explorer-components';
+
+storiesOf('LeaderCard', module).add('LeaderCard', () => (
```
Comment (villanuevawill) [inline src/stories/LeaderCard.stories.js]: :) nice. However, we are not including these in storyboard. Just the general components (this is a specific component)

File: src/public-modules/Leaderboard/index.js [hunk 3]
```diff
@@ -24,6 +26,10 @@ function loadLeaderboardFail(error) {
   return { type: LOAD_LEADERBOARD_FAIL, error };
 }
 
+function useLeaderboard(switchValue = 'Issuers') {
```
Comment (villanuevawill) [inline src/public-modules/Leaderboard/index.js]: more descriptive? toggleLeaderboard

File: src/containers/Leaderboard/LeaderboardHeader.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          Leaderboard
+        </Text>
+      </div>
+      <Switch
+        onValue={'Earners'}
+        offValue={'Issuers'}
+        selectedColor="white"
+        unselectedColor="lightGrey"
+        backgroundColor="purple"
+        switchColor="purpleWhite"
```
Comment (villanuevawill) [inline src/containers/Leaderboard/LeaderboardHeader.js]: just copy what's in storyboard - the colors should match up.

File: src/containers/Leaderboard/LeaderboardHeader.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  return {
+    switchValue: leaderboardState.switchValue
+  };
+};
+
+const LeaderboardHeader = compose(
+  connect(
+    mapStateToProps,
+    {
+      onSwitchChange: actions.useLeaderboard,
```
Comment (villanuevawill) [inline src/containers/Leaderboard/LeaderboardHeader.js]: I'd say it's better to pass function names as to what they're doing. Keep it as toggleLeaderboard vs. onSwitchChange.  That way if someone is passing through the markup quickly, it reads a lot more intuitively.

File: src/stories/LeaderCard.stories.js (hunk 2 above)
Comment (c-o-l-o-r) [reply src/stories/LeaderCard.stories.js]: sorry I meant to remove this, it was just for me to build quickly 🤒

File: src/containers/Leaderboard/LeaderboardHeader.js (hunk 5 above)
Comment (c-o-l-o-r) [reply src/containers/Leaderboard/LeaderboardHeader.js]: good point! thank you

**Comment by c-o-l-o-r** [comment]
I'm closing this because I misspelled leaderboard in the branch name..

**Pull request closed without merging**